File: src/main/scala/com/umatrangolo/giltapi/model/InventoryStatus.scala
package com.umatrangolo.giltapi.model

/**
 * Status of a Sku:
 * - Sold Out: no more inventory for this Sku
 * - For Sale: some still available
 * - Reserved: still available but reserved (10 mins window)
 */
object InventoryStatus extends Enumeration {
  type InventoryStatus = Value

  val SoldOut = Value("sold out")
  val ForSale = Value("for sale")
  val Reserved = Value("reserved")
}

File: src/main/java/com/umatrangolo/giltapi/model/InventoryStatus.java
package com.umatrangolo.giltapi.model;

public enum InventoryStatus {
    SoldOut("sold out"),
    ForSale("for sale"),
    Reserver("reserved");

    private final String key;

    public String getKey() {
        return key;
    }

    InventoryStatus(String key) {
        this.key = key;
    }
}
